### PR TITLE
[v3.6.0] CSM panel modification

### PR DESCRIPTION
**Pull request by troublemaker52025**
Re: #

### Changelog

* 

-------

### Continuous Integration

This pull request:

* [x] needs automatic test cases check.

  > Manual trigger with `@cocos-robot run test cases` afterward.

-------

### Compatibility Check

This pull request:

* [x] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [x] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Comment by troublemaker52025** [comment]
@cocos-robot run test cases

File: cocos/core/renderer/scene/directional-light.ts [hunk 1]
```diff
@@ -52,7 +52,7 @@ export class DirectionalLight extends Light {
     protected _shadowSaturation = 1.0;
     protected _shadowDistance = 50;
     protected _shadowInvisibleOcclusionRange = 200;
-    protected _csmLevel = CSMLevel.level_3;
+    protected _csmLevel = CSMLevel.level_4;
     protected _shadowCSMValueDirty = false;
```
Comment (jk20012001) [inline cocos/core/renderer/scene/directional-light.ts]: _csmValueDirty

File: cocos/core/renderer/scene/shadows.ts [hunk 2]
```diff
@@ -162,6 +162,27 @@ export const CSMLevel = Enum({
...
+    level_1: CSMLevel.level_1,
+
+    /**
+     * @zh 4 个层级
+     * @en level 4
+     * @readonly
+     */
+    level_4: CSMLevel.level_4,
+});
+
```
Comment (jk20012001) [inline cocos/core/renderer/scene/shadows.ts]: UI显示为No CSM和4 Levels CSM

File: cocos/core/scene-graph/scene-globals.ts [hunk 3]
```diff
@@ -809,9 +809,6 @@ export class ShadowsInfo {
     get shadowMapSize () {
         return this._size.x;
     }
-    get size () : Readonly<Vec2> {
-        return this._size;
```
Comment (troublemaker52025) [inline cocos/core/scene-graph/scene-globals.ts:852]: 该函数引擎内部及编辑器已无人使用，故废弃该接口，并添加废弃申明

File: native/cocos/renderer/pipeline/shadow/CSMLayers.cpp [hunk 4]
```diff
@@ -183,7 +183,7 @@ void CSMLayers::update(const PipelineSceneData *sceneData, const scene::Camera *
     if (dirLight->isShadowFixedArea()) {
         updateFixedArea(dirLight);
     } else {
-        if (dirLight->isShadowCSMValueDirty() || _levelCount != levelCount ||
+        if (dirLight->isCsmValueDirty() || _levelCount != levelCount ||
```
Comment (jk20012001) [inline native/cocos/renderer/pipeline/shadow/CSMLayers.cpp]: isCSMValueDirty

File: native/cocos/renderer/pipeline/shadow/CSMLayers.cpp [hunk 5]
```diff
@@ -238,7 +238,7 @@ void CSMLayers::splitFrustumLevels(scene::DirectionalLight *dirLight) {
     // numbers of level - 1
     _layers[level - 1]->setSplitCameraFar(fd);
 
-    dirLight->setShadowCSMValueDirty(false);
+    dirLight->setCsmValueDirty(false);
```
Comment (jk20012001) [inline native/cocos/renderer/pipeline/shadow/CSMLayers.cpp]: setCSMValueDirty

File: native/tools/tojs/scene.ini [hunk 6]
```diff
@@ -99,7 +99,7 @@ getter_setter = Root::[device:_device mainWindow curWindow tempWindow windows sc
        Node::[uuid angle matrix hasChangedFlags/getChangedFlags/setChangedFlags active/isActive _persistNode/isPersistNode/setPersistNode _dirtyFlags/getDirtyFlag/setDirtyFlag],
        Ambient::[skyColor skyIllum groundAlbedo enabled/isEnabled mipmapCount],
        Light::[baked/isBaked/setBaked color useColorTemperature/isUseColorTemperature/setUseColorTemperature colorTemperature colorTemperatureRGB node type name scene],
-       DirectionalLight::[direction illuminance illuminanceHDR illuminanceLDR shadowEnabled/isShadowEnabled shadowPcf shadowBias shadowNormalBias shadowSaturation shadowDistance shadowInvisibleOcclusionRange csmLevel csmLayerLambda csmOptimizationMode shadowFixedArea/isShadowFixedArea shadowNear shadowFar shadowOrthoSize],
+       DirectionalLight::[direction illuminance illuminanceHDR illuminanceLDR shadowEnabled/isShadowEnabled shadowPcf shadowBias shadowNormalBias shadowSaturation shadowDistance shadowInvisibleOcclusionRange csmLevel csmValueDirty/isCsmValueDirty/setCsmValueDirty csmLayerLambda csmOptimizationMode shadowFixedArea/isShadowFixedArea shadowNear shadowFar shadowOrthoSize],
```
Comment (jk20012001) [inline native/tools/tojs/scene.ini]: CSM

**Review by jk20012001 — APPROVED** [review]

File: cocos/3d/lights/directional-light-component.ts [hunk 7]
```diff
@@ -266,26 +266,23 @@ export class DirectionalLight extends Light {
...
+    @type(VisibleLevel)
     get csmLevel () {
         return this._csmLevel;
     }
     set csmLevel (val) {
         this._csmLevel = val;
         if (this._light) {
             this._light.csmLevel = this._csmLevel;
-            this._light.shadowCSMValueDirty = true;
+            this._light.csmValueDirty = true;
```
Comment (pandamicro) [inline cocos/3d/lights/directional-light-component.ts]: What is CSM value ? Please choose a better meaningful name

File: cocos/core/renderer/scene/directional-light.ts [hunk 8]
```diff
@@ -206,11 +206,11 @@ export class DirectionalLight extends Light {
         this._csmLevel = val;
     }
 
-    get shadowCSMValueDirty () {
-        return this._shadowCSMValueDirty;
+    get csmValueDirty () {
```
Comment (pandamicro) [inline cocos/core/renderer/scene/directional-light.ts]: API documentation

File: cocos/core/renderer/scene/shadows.ts [hunk 9]
```diff
@@ -163,7 +163,28 @@ export const CSMLevel = Enum({
 });
 
 /**
- * @zh 联级阴影性能优化模式。
+ * @zh 级联阴影贴图层级。
+ * @en The CSM mode for editor
+ * @enum Shadows.VisibleLevel
+ */
+export const VisibleLevel = Enum({
```
Comment (pandamicro) [inline cocos/core/renderer/scene/shadows.ts]: ShadowMode is better, in CSM, user could specify level for different environment to have better performance

File: cocos/core/renderer/scene/shadows.ts [hunk 10]
```diff
@@ -163,7 +163,28 @@ export const CSMLevel = Enum({
...
+    /**
+     * @zh 1 个层级
+     * @en level 1
+     * @readonly
+     */
+    none: CSMLevel.level_1,
```
Comment (pandamicro) [inline cocos/core/renderer/scene/shadows.ts]: all enum key should be upper cased: NONE

File: cocos/core/renderer/scene/shadows.ts [hunk 11]
```diff
@@ -163,7 +163,28 @@ export const CSMLevel = Enum({
...
+
+    /**
+     * @zh 4 个层级
+     * @en level 4
+     * @readonly
+     */
+    csm: CSMLevel.level_4,
```
Comment (pandamicro) [inline cocos/core/renderer/scene/shadows.ts]: CSM

File: cocos/core/root.ts [hunk 12]
```diff
@@ -41,6 +41,7 @@ import { Pipeline, PipelineRuntime } from './pipeline/custom/pipeline';
 import { createCustomPipeline } from './pipeline/custom';
 import { Batcher2D } from '../2d/renderer/batcher-2d';
 import { IPipelineEvent } from './pipeline/pipeline-event';
+import { SERVER_MODE } from './default-constants';
```
Comment (pandamicro) [inline cocos/core/root.ts]: where does this changes come from, shouldn't be modified

File: editor/assets/chunks/cc-shadow.chunk [hunk 13]
```diff
@@ -1,5 +1,6 @@
 // Copyright (c) 2017-2020 Xiamen Yaji Software Co., Ltd.
 
+#pragma define NUMCASCADES 4
```
Comment (pandamicro) [inline editor/assets/chunks/cc-shadow.chunk:3]: Shouldn't be fixed to 4, must consider how user setup different csm level

File: native/cocos/renderer/pipeline/Define.h [hunk 14]
```diff
@@ -447,6 +446,7 @@ struct CC_DLL UBOCamera {
 };
 
 struct CC_DLL UBOShadow {
+    static constexpr uint CSM_LEVEL_COUNT = 4;
```
Comment (pandamicro) [inline native/cocos/renderer/pipeline/Define.h:447]: Again, shouldn't be fixed

**Review by pandamicro — CHANGES_REQUESTED** [review]
How can user specify Cascade Shadow Map quality, for example use 2 levels or 3 levels csm ?
I suggest adding a global setting in SceneGlobals.ShadowsInfo

File: cocos/core/renderer/scene/directional-light.ts (hunk 8 above)
Comment (troublemaker52025) [reply cocos/core/renderer/scene/directional-light.ts:209]: * https://github.com/cocos/3d-tasks/issues/12510

File: cocos/core/root.ts (hunk 12 above)
Comment (troublemaker52025) [reply cocos/core/root.ts]: come from web CI error, No one has handled this error report at present

File: editor/assets/chunks/cc-shadow.chunk (hunk 13 above)
Comment (troublemaker52025) [reply editor/assets/chunks/cc-shadow.chunk:3]: Level 4 is the maximum number of levels，user cant change this define

File: cocos/core/renderer/scene/shadows.ts (hunk 9 above)
Comment (troublemaker52025) [reply cocos/core/renderer/scene/shadows.ts]: enable csm level 1 or level 4, conceal level 2 and level 3

**Review by pandamicro — APPROVED** [review]

**Review by pandamicro — APPROVED** [review]

**Comment by pandamicro** [comment]
@cocos-robot run test cases

**Pull request merged**